keps/sig-apps/0003-kubernetes-application-api.md

```diff
@@ -0,0 +1,1108 @@
+        app: wordpress
+        component: rdbms
+    spec:
+     
+      containers:
+      - image: mysql:5.6
+        name: mysql
+        env:
+        - name: MYSQL_ROOT_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: mysql-pass
+              key: password
+        ports:
+        - containerPort: 3306
+          name: mysql
+        volumeMounts:
+        - name: mysql-persistent-storage
+          mountPath: /var/lib/mysql
+   volumeClaimTemplates:
+     - metadata:
+         name: mysql-persistent-storage
+       spec:
+         accessModes: [ "ReadWriteOnce" ]
+         resources:
+           requests:
+             storage: 250Gi
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: wordpress-webserver-svc
+  labels:
+    app: wordpress
+    component: wordpress-webserver-svc
+  annotations:
+      kubernetes.io/application: wordpress
+spec:
+  ports:
+    - port: 80
+  selector:
+    app: wordpress
+    component: wordpress-webserver
+  type: LoadBalancer
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: wordpress-webserver-hsvc
+  labels:
+    app: wordpress
+    component: wordpress-webserver-hsvc
+  annotations:
+      kubernetes.io/application: wordpress
+spec:
+  ports:
+    - port: 3306
+  selector:
+    app: wordpress
+    component: wordpress-webserver
+  clusterIP: None
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: wordpress-webserver
+  labels:
+    app: wordpress
+    component: wordpress-webserver
+  annotations:
+      kubernetes.io/application: wordpress
+spec:
+  replicas: 1
+  service: wordpress-webserver-hsvc
+  selector:
+    matchLabels:
+      app: wordpress
+      component: wordpress-webserver
+  template:
+    metadata:
+      labels:
+        app: wordpress
+        tier: frontend
+    spec:
+      containers:
+      - image: wordpress:4.8-apache
+        name: wordpress
+        env:
+        - name: WORDPRESS_DB_HOST
+          value: wordpress-mysql
+        - name: WORDPRESS_DB_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: mysql-pass
+              key: password
+        ports:
+        - containerPort: 80
+          name: wordpress
+        volumeMounts:
+        - name: wordpress-persistent-storage
+          mountPath: /var/www/html
+  volumeClaimTemplates:
+       - metadata:
+           name: mysql-persistent-storage
+         spec:
+           accessModes: [ "ReadWriteOnce" ]
+           resources:
+             requests:
+               storage: 250Gi
+```
+
+The components of the WordPress application are made explicit by the `Spec.Components` of the `word-press` Application.
+When clients examine the `Spec.Components` of the Application, they have enough information to retrieve and display 
+any of the Kubernetes primitives that compose the Application. When any of those components is examined, clients can use 
+the name indicated by the `kubernetes.io/application` to retrieve the parent Application object.
```

Comment:
I understand the point of this piece of information, but can we avoid the extra-redundancy? I see below that clients should discard objects that don't have this annotation. How do they learn about this error in their configuration file? Beside conflicts (in the case of two apps trying to own the same object), what's preventing the application controller from adding the annotation?